### PR TITLE
mgr/dashboard: Filter out tasks depending on permissions

### PR DESCRIPTION
**Pull request by Tinatinak**
Fixes: https://tracker.ceph.com/issues/25094

Signed-off-by: Tina Kallio <tina.kallio@gmail.com>

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

**Comment by s0nea** [comment]
jenkins test make check arm64

File: src/pybind/mgr/dashboard/controllers/summary.py [hunk 1]
```diff
@@ -41,15 +41,43 @@ def _rbd_mirroring(self):
...
 
+    def _task_permissions(self, name):
+        result = True
+        if name == 'pool/create':
+            result = self._has_permissions(Permission.CREATE, Scope.POOL)
+        elif name == 'pool/edit':
+            result = self._has_permissions(Permission.UPDATE, Scope.POOL)
+        elif name == 'pool/delete':
+            result = self._has_permissions(Permission.DELETE, Scope.POOL)
+        elif name in [
```
Comment (rjfd) [inline src/pybind/mgr/dashboard/controllers/summary.py]: The required permissions for each type of task should match what is in the original declaration of the controller that creates the task. The `rbd/*` tasks are created/declared in the `controllers/rbd.py` file. 

For instance, the `rbd/flatten` requires update permission and not create.

Comment (sebastian-philipp) [reply src/pybind/mgr/dashboard/controllers/summary.py]: How can we make sure, this is not getting outdated in the future?

Comment (rjfd) [reply src/pybind/mgr/dashboard/controllers/summary.py]: I think the only way would be to encode the required permissions inside the task object upon task creation.

But that requires a much more complex solution.

Comment (Tinatinak) [reply src/pybind/mgr/dashboard/controllers/summary.py]: Changes made: task types now match the required permissions.

File: src/pybind/mgr/dashboard/controllers/summary.py [hunk 2]
```diff
@@ -41,15 +41,46 @@ def _rbd_mirroring(self):
...
+                'rbd/delete', 'rbd/snap/delete', 'rbd/trash/move',
+                'rbd/trash/remove', 'rbd/trash/purge']:
+            result = self._has_permissions(Permission.DELETE, Scope.RBD_IMAGE)
+        return result
+
     @Endpoint()
     def __call__(self):
         executing_t, finished_t = TaskManager.list_serializable()
+        executing_tasks = []
```
Comment (sebastian-philipp) [inline src/pybind/mgr/dashboard/controllers/summary.py]: minor nit:

```python
executing_tasks = [task for task in executing_tasks if self._task_permissions(task['name'])]

Comment (Tinatinak) [reply src/pybind/mgr/dashboard/controllers/summary.py]: Thanks for the tip! I have changed now the for-loops accordingly.

**Pull request merged**